packages/core/test/oracle/VotingV2.js

```diff
@@ -2331,6 +2331,81 @@ describe("VotingV2", function () {
+
+    // State variables should be set correctly.
+    assert.equal((await voting.methods.voterStakes(account1).call()).delegateVoting, rand);
+    assert.equal(await voting.methods.delegateToStaker(rand).call(), account1);
+
+    // Request a price and see that we can vote on it on behalf of the account1 from the delegate.
+    const identifier = padRight(utf8ToHex("delegated-voting"), 64); // Use the same identifier for both.
+    const time = "420";
+    await supportedIdentifiers.methods.addSupportedIdentifier(identifier).send({ from: accounts[0] });
+    await voting.methods.requestPrice(identifier, time).send({ from: registeredContract });
+
+    await moveToNextRound(voting, accounts[0]);
+    const roundId = (await voting.methods.getCurrentRoundId().call()).toString();
+
+    const price = 1;
+
+    const salt = getRandomSignedInt(); // use the same salt for all votes. bad practice but wont impact anything.
+
+    // construct the vote hash. Note the account is the delegate.
+    const hash = computeVoteHash({ salt, roundId, identifier, price, account: rand, time });
+
+    // Commit the votes.
+    await voting.methods.commitVote(identifier, time, hash).send({ from: rand });
+    await moveToNextPhase(voting, accounts[0]); // Reveal the votes.
+    await voting.methods.revealVote(identifier, time, price, salt).send({ from: rand });
+    await moveToNextRound(voting, accounts[0]); // Move to the next round.
+
+    // The price should have settled as per usual and be recoverable. The original staker should have gained the positive
+    // slashing from being the oly correct voter. The total slashing should be 68mm * 0.0016 = 108800.
+    assert.equal(await voting.methods.getPrice(identifier, time).call({ from: registeredContract }), price);
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    assert.equal(
+      (await voting.methods.voterStakes(account1).call()).cumulativeStaked,
+      toWei("32000000").add(toWei("108800"))
+    );
+  });
+
+  it("Existing stakers cant become delegates and delegates cant stake", async function () {
+    // Account1 has a staked balance so cant be used by account2 in delegation.
+    assert(await didContractThrow(voting.methods.delegateVoting(account1).send({ from: account2 })));
+
+    // Before the user accepts  the delegation they can stake but after they accept it they can no longer stake.
+    await voting.methods.delegateVoting(rand).send({ from: account1 });
+    await votingToken.methods.mint(rand, toWei("100")).send({ from: accounts[0] });
+    await votingToken.methods.approve(voting.options.address, toWei("100")).send({ from: rand });
+
+    // Should be able to stake before accepting the delegation.
+    await voting.methods.stake(toWei("50")).send({ from: rand });
+
+    // After accepting the delegation can no longer stake.
+    await voting.methods.acceptDelegation(account1).send({ from: rand });
```

Comment:
I'm wondering what could be the use case of allowing an `acceptDelegation` if the user stakes between `delegateVoting` and `acceptDelegation`

I notice that when a `rand` user accepts delegation, he or she can no longer vote in his or her own name right?
From there `rand` has three options:
- He delegates his vote to someone else to be able to vote on his/her behalf
- He doesn't vote and get's slashed
- acceptDelegation & unstake